pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java

```diff
@@ -32,6 +36,51 @@ public class DataTypeConversionFunctions {
   private DataTypeConversionFunctions() {
   }
 
+  @ScalarFunction(nullableParameters = true)
+  public static Object cast(Object value, String targetTypeLiteral) {
+    if (value == null || targetTypeLiteral == null) {
+      return value;
+    }
+    try {
+      Class<?> clazz = value.getClass();
+      boolean isMultiValue = clazz.isArray() & clazz != byte[].class;
+      PinotDataType sourceType = isMultiValue
+          ? PinotDataType.getMultiValueType(clazz)
```

Comment:
We need to read the class for the first element in order to get the correct type. We might also need to differentiate the primitive value array and boxed value array (seems the input value is always primitive value array)

Reply:
Dropping MV support.